### PR TITLE
Sagemaker Tasks ported

### PR DESCRIPTION
**Pull request by kumare3**

### CODE REVIEW

File: flytekit/taskplugins/sagemaker/hpo.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        return MessageToDict(_hpo_job_model.HyperparameterTuningJob(
+            max_number_of_training_jobs=self._hpo_job_config.max_number_of_training_jobs,
+            max_parallel_training_jobs=self._hpo_job_config.max_parallel_training_jobs,
+            training_job=training_job,
+        ).to_flyte_idl())
+
+
+# %%
+# HPO Task allows ParameterRangeOneOf and HyperparameterTuningJobConfig as inputs. In flytekit this is possible
+# to allow these two types to be registered as valid input / output types and provide a custom transformer
```
Comment (katrogan) [inline flytekit/taskplugins/sagemaker/hpo.py:77]: this is so cool!

File: tests/flytekit/unit/taskplugins/sagemaker/test_training.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import pytest
+
+from flytekit import task, metadata, kwtypes
+from flytekit.annotated.context_manager import RegistrationSettings, Image, ImageConfig
+from flytekit.taskplugins.sagemaker import SagemakerTrainingJobConfig, SagemakerBuiltinAlgorithmsTask, \
+    TrainingJobResourceConfig, AlgorithmSpecification, AlgorithmName
+from flytekit.types import FlyteFile
+
+
+def _get_reg_settings():
```
Comment (katrogan) [inline tests/flytekit/unit/taskplugins/sagemaker/test_training.py:14]: maybe we should have a testutils and move this there, considering the amount of times we create a RegistrationSettings obj for tests :)

**Review by katrogan — APPROVED** [review]
the flytekit stuff looks fine to me but i don't really understand anything related to sagemaker

File: tests/flytekit/unit/taskplugins/sagemaker/test_training.py (hunk 2 above)
Comment (kumare3) [reply tests/flytekit/unit/taskplugins/sagemaker/test_training.py:21]: ya

**Pull request merged**